src/main/kotlin/ai/digital/integration/server/deploy/internals/cluster/operator/AwsEksHelper.kt

```diff
@@ -1,24 +1,391 @@
+        }
+    }
+
+    private fun existingStack(): Boolean {
+        val stackStatusCmd = "aws --region ${getProvider().region.get()} " +
+                "cloudformation  describe-stacks" +
+                " --stack-name " +
+                "${getProvider().stack.get()} " +
+                "--query 'Stacks[0].StackStatus' " +
+                "--output text"
+
+        val result = ProcessUtil.executeCommand(project,
+                "$stackStatusCmd",
+                logOutput = true,
+                throwErrorOnFailure = false)
+        if (result == "CREATE_COMPLETE") {
+            return true
+        } else if (result.contains("does not exist")) {
+            return false
+        }
+        return false
+    }
+
+
+    private fun verifyClusterStatus(totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val awsEksProvider: AwsEksProvider = getProvider()
+        stackStatus(awsEksProvider, totalTimeInSec, sleepTime, stackId, status)
+    }
+
+    private fun stackStatus(awsEksProvider: AwsEksProvider, totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val stackStatusCmd = "aws --region ${awsEksProvider.region.get()} " +
+                "cloudformation  list-stacks " +
+                "--query 'StackSummaries[?StackId==`$stackId`].StackStatus' " +
+                "--output text"
+        val stackStatus = wait(status,
+                stackStatusCmd,
+                "Stack",
+                totalTimeInSec,
+                sleepTime)
+        if (!stackStatus) {
+            throw RuntimeException("Resource STACK status is not $status")
+        }
+    }
+
+    private fun wait(status: String, command: String, resource: String, totalTimeInSec: Int = 3000, sleepTime: Long = 1000): Boolean {
+        val expectedEndTime = System.currentTimeMillis() + totalTimeInSec // 20 mins
+        while (expectedEndTime > System.currentTimeMillis()) {
+            val result = ProcessUtil.executeCommand(project,
+                    "$command",
+                    logOutput = true,
+                    throwErrorOnFailure = false)
+            if (result.contains("$status")) {
+                return true
+            } else if (result.contains("does not exist")) {
+                return false
+            }
+            project.logger.lifecycle("$resource resource  \"$status\" status not met, retry. ")
+            Thread.sleep(sleepTime) //5 mins
+        }
+        return false
+    }
+
+    private fun createStorageClass(efsStorageClassName: String) {
+        createStorageClassEFS(efsStorageClassName)
+        getKubectlHelper().setDefaultStorageClass("gp2", efsStorageClassName)
+    }
+
+    private fun createStorageClassEFS(storageClassName: String) {
+        if (!getKubectlHelper().hasStorageClass(storageClassName)) {
+            project.logger.lifecycle("Create storage class: {}", storageClassName)
+            val fileSystemId = getFileSystemId()
```

Comment:
Why do we need a variable for that?
What's the problem to pass `getFileSystemId()` directly to `helmInstallAwsEfs` as a parameter?